pkg/file/error.go

```diff
@@ -45,7 +45,29 @@ func (e *HashError) Unwrap() error {
 	return e.err
 }
 
-// Error implement standard go error interface.
+// Error implements standard go error interface.
 func (e *HashError) Error() string {
 	return e.err.Error()
 }
+
+// SplitError should wrap any error returned from the splitter.
```

Comment:
Split error type is used here to wrap other errors, but I do not see that it is checked anywhere, it is not used. I think that it is not needed until we explicitly handle it. I suggest to remove it and use fmt.Errorf to wrap errors, just as it it was.